libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -374,152 +399,91 @@ void AC_WPNav::get_wp_stopping_point(Vector3f& stopping_point) const
 /// advance_wp_target_along_track - move target location along track from origin to destination
 bool AC_WPNav::advance_wp_target_along_track(float dt)
 {
-    float track_covered;        // distance (in cm) along the track that the vehicle has traveled.  Measured by drawing a perpendicular line from the track to the vehicle.
-    Vector3f track_error;       // distance error (in cm) from the track_covered position (i.e. closest point on the line to the vehicle) and the vehicle
-    float track_desired_max;    // the farthest distance (in cm) along the track that the leash will allow
-    float track_leash_slack;    // additional distance (in cm) along the track from our track_covered position that our leash will allow
-    bool reached_leash_limit = false;   // true when track has reached leash limit and we need to slow down the target point
-
-    // get current location
-    const Vector3f &curr_pos = _inav.get_position();
-
     // calculate terrain adjustments
     float terr_offset = 0.0f;
     if (_terrain_alt && !get_terrain_offset(terr_offset)) {
         return false;
     }
 
-    // calculate 3d vector from segment's origin
-    Vector3f curr_delta = (curr_pos - Vector3f(0,0,terr_offset)) - _origin;
-
-    // calculate how far along the track we are
-    track_covered = curr_delta.x * _pos_delta_unit.x + curr_delta.y * _pos_delta_unit.y + curr_delta.z * _pos_delta_unit.z;
-
-    // calculate the point closest to the vehicle on the segment from origin to destination
-    Vector3f track_covered_pos = _pos_delta_unit * track_covered;
+    // get current position and adjust altitude to origin and destination's frame (i.e. _frame)
+    const Vector3f &curr_pos = _inav.get_position() - Vector3f(0, 0, terr_offset);
 
-    // calculate the distance vector from the vehicle to the closest point on the segment from origin to destination
-    track_error = curr_delta - track_covered_pos;
+    // Adjust time to prevent target moving too far in front of aircraft
+//    const Vector3f &curr_vel = _inav.get_velocity();
```

Comment:
why the commented code?

Reply:
I've removed the commented out code and replaced it with a ToDo.  In the PR comments you'll see there is this known issue: Target point does not slow if vehicle cannot keep up